valhalla/mjolnir/osmpbfparser.h

```diff
@@ -34,10 +34,13 @@
 #include <fstream>
 #include <string>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
 // this describes the low-level blob storage
 #include <valhalla/proto/fileformat.pb.h>
 // this describes the high-level OSM objects
 #include <valhalla/proto/osmformat.pb.h>
+#pragma GCC diagnostic pop
```

Comment:
So my local protobuf has plenty of unused parameters. I see at least to ways here
1. Disable specific warnings for thirparty/external headers. This option is implemented in the PR
2. Disable specific warnings across all project. e.g. https://github.com/mapbox/mapbox-gl-native/blob/b8edc2399b9640498ccbbbb5b8f058c63d070933/CMakeLists.txt#L61